### PR TITLE
Feature/bump netlink#main

### LINKED ISSUE #43551: Docker Engine 20.10.14 breaks with debian kernel `linux-image-5.10.0-13-amd64`
**Description**

**Steps to reproduce the issue:**
1. Install 2x Debian 11.3 with Docker 20.10.14
2. Create Swarm
3. Create encrypted overlay network
4. Launch a dummy webserver container on node A
5. Launch another container on node B which can do http request against webserver on node A
6. Expect: I/O Request Timout

1b. Create NOT-encrypted overlay network
2c. Launch a dummy webserver container on node A
3d. Launch another container on node B which can do http request against webserver on node A
4e. Expect: B receive answer from A

**Describe the results you received:**
I/O Request Timout

**Describe the results you expected:**
Containers able to talk to each other

**Additional information you deem important (e.g. issue happens only occasionally):**
I did not test with in between version Kernel 5.10.0-12-amd64. 
**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           20.10.14
 API version:       1.41
 Go version:        go1.16.15
 Git commit:        a224086
 Built:             Thu Mar 24 01:48:21 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.14
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.15
  Git commit:       87a90dc
  Built:            Thu Mar 24 01:46:14 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.11
  GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0


```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.8.1-docker)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 45
  Running: 44
  Paused: 0
  Stopped: 1
 Images: 349
 Server Version: 20.10.14
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: x0rht2cqbs5n45g106ah6jj3y
  Is Manager: true
  ClusterID: 129c31wk0v99pe97fx0fegehw
  Managers: 3
  Nodes: 3
  Default Address Pool: 10.0.0.0/8  
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 45.9.60.19
  Manager Addresses:
   45.142.177.91:2377
   45.9.60.124:2377
   45.9.60.19:2377
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc version: v1.0.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.0-11-amd64
 Operating System: Debian GNU/Linux 11 (bullseye)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.63GiB
 Name: marauder.intra6.de
 ID: MVAU:BGKZ:NGFI:TSVO:UDB2:ZQB2:DHFM:TOPV:XQLI:2WWX:SGQ5:Z7T6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
  provider=generic
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false


```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
netcup.de root-vServer kvm

### PR DESCRIPTION
**Pull request by s4ke**
Attempting to fix #43551

Will update according to contribution guidlines if the fix is fine.

**- What I did**

Bumped netlink dependencies to newest release (currently in beta)

**- How to verify it**

still in progress, but if everyone agrees that this is the correct course of action, I will verify this on currently affected servers.

**- Description for the changelog**

Bump netlink dependency to fix issues where encrypted overlay networks did not work with newer Linux kernel releases.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by s4ke** [comment]
Since I am new to working on this project, I would greatly appreciate some input on these:

0. Is moby/moby the correct repo for this fix or should I have done this in libnetwork?
1. Can I test the encryption behaviour with two docker daemons running in dind or do I have to spin up VMs for this?
2. Did I manipulate vendor.mod correctly?

**Comment by s4ke** [comment]
closing this as the issue seems to be more complex

File: vendor.mod [hunk 1]
```diff
@@ -177,7 +177,7 @@ replace (
 	github.com/matttproud/golang_protobuf_extensions => github.com/matttproud/golang_protobuf_extensions v1.0.1
 	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.6.0
 	github.com/prometheus/procfs => github.com/prometheus/procfs v0.0.11
-	github.com/vishvananda/netlink => github.com/vishvananda/netlink v1.1.0
+	github.com/vishvananda/netlink => github.com/vishvananda/netlink v1.2.1-beta.2
```
Comment (thaJeztah) [inline vendor.mod:180]: We can remove this replace rule if we decide to update to a new release; ISTR we added this to not update to a non-released version if there was no need. As there's a (beta) release now, perhaps we should consider updating to a newer version either way 🤔

**Comment by thaJeztah** [comment]
Yes, I think it'd be ok to update the dependency to a more current version; if you have time to work on updating the PR; please do 👍 (please amend the existing commit so that there's only a single commit in the PR - but happy to help with that if needed!)

**Comment by s4ke** [comment]
locally the tests are failing for me, so I will have to check this out further:

```
=== Skipped
=== SKIP: libnetwork/iptables TestFirewalldInit (0.00s)
    firewalld_test.go:14: firewalld is not running

=== Failed
=== FAIL: libnetwork/drivers/bridge TestPortMappingV6Config (0.12s)
time="2022-06-16T17:32:48Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge/dummy/ is not added to the store"
time="2022-06-16T17:32:48Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep1/ is not added to the store"
time="2022-06-16T17:32:48Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep1/ is not added to the store"
    port_mapping_test.go:163: Failed to store the port bindings into the sandbox info. Found: [{udp 172.19.0.11 400 0.0.0.0 54000 54000} {tcp 172.19.0.11 500 0.0.0.0 65000 65000} {sctp 172.19.0.11 500 0.0.0.0 65000 65000}]

=== FAIL: libnetwork/ipam TestParallelPredefinedRequest2 (0.00s)
panic: send on closed channel [recovered]
        panic: send on closed channel

goroutine 27 [running]:
testing.tRunner.func1.2({0x64b7c0, 0x6dfb10})
        /usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x64b7c0, 0x6dfb10})
        /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/docker/docker/libnetwork/ipam.runParallelTests(0xc000503520, 0x1)
        /go/src/github.com/docker/docker/libnetwork/ipam/allocator_test.go:1503 +0x28b
github.com/docker/docker/libnetwork/ipam.TestParallelPredefinedRequest2(0x5ed747?)
        /go/src/github.com/docker/docker/libnetwork/ipam/allocator_test.go:1607 +0x1e
testing.tRunner(0xc000503520, 0x6a2390)
        /usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1486 +0x35f

=== FAIL: libnetwork/ipam TestParallelPredefinedRequest4 (unknown)

=== FAIL: libnetwork/ipam TestParallelPredefinedRequest1 (unknown)

=== FAIL: libnetwork/ipam TestParallelPredefinedRequest3 (unknown)

DONE 326 tests, 1 skipped, 5 failures in 202.256s
make: *** [Makefile:235: test-unit] Error 1
```

**Comment by thaJeztah** [comment]
Let's see what CI does; I know we inherited some, erm, "weird" tests from libnetowrk, some of which are not-so-very-unit-tests that may require some setting up, so perhaps they're ok.

File: vendor.mod (hunk 1 above)
Comment (thaJeztah) [inline vendor.mod:180]: oh! I should've looked at the diff; could you remove this `replace` (per the earlier discussion?)

**Comment by thaJeztah** [comment]
(fwiw; don't put it as a separate `require` line, but move it into the first `require` section if the script doesn't automatically put it there)

File: vendor.mod (hunk 1 above)
Comment (s4ke) [reply vendor.mod:180]: Yeah sure. I could have sworn I did that.

**Comment by s4ke** [comment]
I think I got it now. I greatly appreciate the handholding here :). Great first commiter experience I must say :)

**Comment by s4ke** [comment]
CI seems happy with the latest changes :)

**Comment by s4ke** [comment]
Hmm, seems like CI is failing for windows-2022? Tbh, I am having a hard time reading the test logs here on GitHub, but to me these errors look unrelated to the PR?

**Comment by s4ke** [comment]
CI is green :robot:

**Review by thaJeztah — APPROVED** [review]
whoop!

LGTM

**Comment by thaJeztah** [comment]
@tianon @corhere PTAL

**Review by corhere — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thx! Let me bring this one in

**Pull request merged**